### PR TITLE
Schedule manager refactor

### PR DESCRIPTION
**Pull request by Myoldmopar**
Pull request overview
---------------------
It's unclear at this point exactly what all this specific PR will include.  There will be work to clean up the schedule manager, refactor it, add testing to it, and eventually add multi-year scheduling.  This may only be an initial PR to clean up the code before making functional changes that could trigger small diffs.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by nrel-bot-2b** [comment]
@Myoldmopar @lgentile it has been 28 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
I'm expecting custom_check to fail again, with both a missing IDF in the cmake rules and also the PR is missing labels.  But I improved both of those scripts to provide better links and info to the user, after I verify it properly warns with these changes, I'll get rid of those issues and custom_check will pass.  Other, more serious test failures are still a little ways away from being resolved.

File: testfiles/AirflowNetwork_Simple_SmallOffice.idf [hunk 1]
```diff
@@ -1467,7 +1464,10 @@
     For: AllOtherDays,       !- Field 13
     Until: 6:00,1.0,         !- Field 14
     Until: 13:00,0.0,        !- Field 16
-    Until: 24:00,1.0;        !- Field 18
+    Until: 24:00,1.0,        !- Field 18
+    Through: 12/31,          !- Field 1
+    For: AllDays,            !- Field 2
+    Until: 24:00,0.0;        !- Field 3
```
Comment (Myoldmopar) [inline testfiles/AirflowNetwork_Simple_SmallOffice.idf]: This file shows diffs, but it is because this schedule was malformed and needed to be repaired.

File: testfiles/FurnaceWithDXSystem_CoolingHXAssisted.idf [hunk 2]
```diff
@@ -960,7 +957,10 @@
     For: AllOtherDays,       !- Field 13
     Until: 6:00,1.0,         !- Field 14
     Until: 13:00,0.0,        !- Field 16
-    Until: 24:00,1.0;        !- Field 18
+    Until: 24:00,1.0,        !- Field 18
+    Through: 12/31,          !- Field 1
+    For: AllDays,            !- Field 2
+    Until: 24:00,0.0;        !- Field 3
```
Comment (Myoldmopar) [inline testfiles/FurnaceWithDXSystem_CoolingHXAssisted.idf]: This file also shows diffs, and for the same reason, this schedule was completely wrong before.

File: testfiles/AirflowNetwork_Simple_SmallOffice.idf (hunk 1 above)
Comment (mjwitte) [reply testfiles/AirflowNetwork_Simple_SmallOffice.idf]: So, was the remainder of the year zeros before? Would that still be appropriate but with a warning? Just thinking out loud.

Comment (Myoldmopar) [reply testfiles/AirflowNetwork_Simple_SmallOffice.idf]: I considered evaluating what it was doing before, but frankly I believe it is clearly telling the simulation to do two things at a point in the year, so I didn't worry about it.  I intend on just continuing to abort the simulation in this particular case.

**Comment by rraustad** [comment]
Why is this a draft and labeled V9.2 milestone? More changes anticipated?

**Comment by Myoldmopar** [comment]
Yes, @rraustad, this still needs work.  It's an enormous change to the schedule manager, full rewrite, and it needs lots and lots of testing for corner cases related to daylight savings time, repeated run periods, leap years, etc., before getting merged in.  It has a lot of promise for being better with runtime, but needs more work.  I'm hoping to get time to test this heavily next week so it could still go in for the 9.2.0 release, but it's still up in the air.

**Comment by nrel-bot-3** [comment]
@Myoldmopar @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar @lgentile it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 28 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 12 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 11 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar it has been 19 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 7 days since this pull request was last updated.

**Comment by nrel-bot-2c** [comment]
@Myoldmopar it has been 8 days since this pull request was last updated.

**Comment by nrel-bot-2b** [comment]
@Myoldmopar it has been 9 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@Myoldmopar it has been 13 days since this pull request was last updated.

**Comment by Myoldmopar** [comment]
I'm closing this temporarily.  I have a new branch that I'll push shortly that addresses the biggest schedule manager fix -- the file reading.  Then we can come back to this broader refactor later.
